core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -111,6 +111,10 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val KILL_ON_OOM_ERROR = ConfigBuilder("spark.driver.killOnOOMError")
+    .doc("Whether to kill the driver on an oom error in cluster mode.")
+    .booleanConf.createWithDefault(true)
```

Comment:
Please split this into two lines like the other conf.

Reply:
And this should be `false` by default to avoid the behavior change.

Reply:
actually I think we agreed to go ahead and change the default
https://github.com/apache/spark/pull/25229#issuecomment-519186339

3.0 is a good chance to do this

and I agree about splitting to two lines to match the style for other confs

Reply:
Thanks. Got it. In that case, I'm +1 for the `true` by default.

BTW, @squito . Do you think we need to add a migration guide for this behavior change?

Also, cc @gatorsmile .

Reply:
Yes we agreed to change the default, we want to fail by default.